Tests/Adyen Tests/Analytics/AnalyticsEventDataSourceTests.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import XCTest
+@_spi(AdyenInternal) @testable import Adyen
+
+final class AnalyticsEventDataSourceTests: XCTestCase {
+    
+    var sut: AnalyticsEventDataSource!
```

Comment:
In my experience having state inside of tests always leads to weird behavior.
Maybe consider moving this to local state - so you also don't have to setup and tear down